### PR TITLE
Implement tutorial EventManager, fix tutorial FSM bug, and add jest to eslint env

### PR DESCRIPTION
**Pull request by arthurshir**
## Changes
* Implement and test tutorial EventManager
* Fix fsm bug
* Add jest to eslint env

## Testing Done
* Wrote unit tests for tutorial EventManager
* Update tutorial fsm tests to test the case where a user's tutorialState is undefined.

### CODE REVIEW

**Comment by arthurshir** [comment]
- Add jsdocs
- Re: timeout comment
    - either add a line in NONFEATURETODOS
    - or just pre-pend the comment with TODO

**Comment by arthurshir** [comment]
> * Add jsdocs
> * Re: timeout comment
>   
>   * either add a line in NONFEATURETODOS
>   * or just pre-pend the comment with TODO

Or just do jest.advanceTimersByTime(1000);

**Comment by arthurshir** [comment]
Add todo for something along the lines of putting `setTimeout` behind an abstraction like this `fireFunctionsInSequence(ctx.reply(..),1000,ctx.reply(...))`

File: tutorial/eventManager.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    static registerAndReactToEvent(userId, eventId, ctx) {
+        let fromStateId = fsm.getStateForUser(userId);
+        fsm.registerEvent(userId, eventId);
+        let toStateId = fsm.getStateForUser(userId);
+
+        if (fromStateId != toStateId && toStateId in stateToCallback) {
+            let callback = stateToCallback[toStateId];
+            callback(ctx);
+        } else {
+            logger.info(`Tried to react to event for user: "${userId}", but no callback exists for the user's tutorialState: "${toStateId}".`);
```
Comment (arthurshir) [inline tutorial/eventManager.js:90]: will not always be accurate due to `(fromStateId != toStateId` case

File: tst/tutorial/eventManager.test.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        expect(ctx.reply).toHaveBeenCalledWith(REPLY_PARAMS.SET_TIMEZONE.TEXT);
+
+        ctx = {reply: jest.fn()};
+        EventManager.registerAndReactToEvent(USER_1, EVENTS.VALID_TIMEZONE, ctx);
+        expect(ctx.reply).toHaveBeenCalledWith(REPLY_PARAMS.SET_REMINDER.TEXT);
+
+        ctx = {reply: jest.fn()};
+        EventManager.registerAndReactToEvent(USER_1, EVENTS.VALID_REMINDER, ctx);
+        expect(ctx.reply).toHaveBeenCalledWith(REPLY_PARAMS.ENDING_TUTORIAL.TEXT_1);
+        jest.advanceTimersByTime(1000);
```
Comment (bubakazouba) [inline tst/tutorial/eventManager.test.js:65]: use a constant for the delay, it'd be annoying to replace this all over once we change it.

File: tutorial/eventManager.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    SET_TIMEZONE: {
+        TEXT: "set your timezone using /timezone (example: /timezone America Los Angeles, or /timezone PDT). You can also share your location and I'll figure out your timezone for you.",
+    },
+    SET_REMINDER: {
+        TEXT: "Alright, now you are ready to set reminders!.\nTry to set this reminder: <code>remind me in 20m to check on the oven</code>",
+    },
+    ENDING_TUTORIAL: {
+        TEXT_1: "Now you if you don't actually have a lasagna in the oven, use the /list command to list your reminders, select it and press the delete button.",
+        TEXT_2: "Alright, you are all set for using the reminder bot now.\n If you have any questions or feature requests, feel free to reach out to @bubakazouba, I am very responsive on telegram!\nYou can use the /help command for more ways to use this reminder bot, also checkout /help_with_recurring_reminders for recurring reminders!",
+    }
```
Comment (bubakazouba) [inline tutorial/eventManager.js:51]: "Now if you don't have the oven running, use the /list ..."

File: tutorial/eventManager.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        MARKUP: Extra.HTML().markup((m) => {
+            let buttons = [
+                m.callbackButton("Alright!", BOT_ACTIONS.TUTORIAL_ACTION_YES),
+                m.callbackButton("I dont need a tutorial", BOT_ACTIONS.TUTORIAL_ACTION_NO),
+            ];
+            return m.inlineKeyboard(buttons);
+        }),
+    },
+    SET_TIMEZONE: {
+        TEXT: "set your timezone using /timezone (example: /timezone America Los Angeles, or /timezone PDT). You can also share your location and I'll figure out your timezone for you.",
```
Comment (bubakazouba) [inline tutorial/eventManager.js:43]: "Set*

Comment (bubakazouba) [reply tutorial/eventManager.js:43]: Maybe all tutorial text should start with this emoji: ❕
to make it clear that this is different text guiding the user through the tutorial

**Review by bubakazouba — APPROVED** [review]

**Pull request merged**